File: assignment_2/Model/Expressions/IExpression.java
package Model.Expressions;

import ADTs.IDictionary;
import Exceptions.*;

public abstract class IExpression {
    public abstract int eval(IDictionary<String, Integer> table) throws UndefinedVariableException, DivisionByZeroException, UndefinedOperationException; // we put this as a parameter because our expressions may need to look up variable
                                                           // names in the symbol table
    public abstract String toString();
}

File: Toy_Language_Interpreter_C/Expressions/IExpression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace a7
{
    public abstract class IExpression
    {
        abstract public int Eval(IDictionary<String, int> table);
        abstract public override String ToString();
    }
}
